src/App.tsx

```diff
@@ -15,36 +18,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+export enum SortType {
+  ALPHABETICALLY = 'ALPHABETICALLY',
+  BY_LENGTH = 'BY_LENGTH',
+}
+
+export type RereversType = true | false;
+
+interface SortParams {
+  sortType: SortType;
+  isReverse: RereversType;
+}
+
+function getPreparedGoods(goods: Good[], { sortType, isReverse }: SortParams) {
+  const preparedGoods = [...goods];
+
+  preparedGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case 'ALPHABETICALLY':
+        return good1.localeCompare(good2);
+
+      case 'BY_LENGTH':
+        return good1.length - good2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReverse, setIsReverse] = useState(false);
+  const [sortType, setSortType] = useState(SortType.ALPHABETICALLY);
+
+  const sortedGoods = getPreparedGoods(goodsFromServer, {
+    sortType,
+    isReverse,
+  });
+
+  function sortReset() {
+    setSortType(SortType.ALPHABETICALLY);
+    setIsReverse(false);
+  }
```

Comment:
```suggestion
  function sortReset() {
    setSortType(SortType.Default);
    setIsReverse(false);
  }
```
create the default field in the enum